src/js/jsx/search/SearchBar.jsx

```diff
@@ -88,50 +91,188 @@ define(function (require, exports, module) {
                 layers = document.layers.allVisible.reverse(),
                 layerMap = layers.map(function (layer) {
                     // Used to determine the layer face icon
-                    var iconID = "layer-";
-                    if (layer.isArtboard) {
-                        iconID += "artboard";
-                    } else if (layer.kind === layer.layerKinds.BACKGROUND) {
-                        iconID += layer.layerKinds.PIXEL;
-                    } else if (layer.kind === layer.layerKinds.SMARTOBJECT && layer.isLinked) {
-                        iconID += layer.kind + "-linked";
-                    } else {
-                        iconID += layer.kind;
-                    }
-                    
-                    return { id: "layer_" + layer.id.toString(), title: layer.name, type: "item", svgType: iconID };
-                }),
-                layerLabel = { id: "layer_header", title: "Layers", type: "header" },
+                    var iconID = this._getSVGInfo(layer),
+                        ancestry = this._formatLayerAncestry(layer);
+
+                    return {
+                        id: "layer_" + layer.id.toString(),
+                        title: layer.name,
+                        info: ancestry,
+                        displayInfo: ancestry,
+                        svgType: iconID,
+                        type: "item"
+                    };
+                }.bind(this)),
+
+                // Get shortest unique layer ancestry
+                ancestors = collection.pluck(layerMap, "info"),
+                shortenedPaths = pathUtil.getShortestUniquePaths(ancestors.toArray());
+
+            layerMap.map(function (layerItem, index) {
+                var newInfo = shortenedPaths[index];
+                
+                // Only show ancestry if other layers have the same name
+                if (layerItem.title === newInfo) {
+                    layerItem.displayInfo = "";
+                } else {
+                    layerItem.displayInfo = newInfo;
+                }
+            });
+
+            var layerLabel = {
+                    id: "layer_header",
+                    title: "Layers",
+                    type: "header"
+                },
                 layerOptions = layerMap.unshift(layerLabel);
 
-            // get list of currently open documents
-            var docStore = this.getFlux().store("document"),
+            return layerOptions;
+        },
+
+        /**
+         * Make list of currently open documents to be used as dropdown options
+         * 
+         * @return {Array.<Object>}
+         */
+        _getCurrDocOptions: function () {
+            var appStore = this.getFlux().store("application"),
+                docStore = this.getFlux().store("document"),
+                document = appStore.getCurrentDocument(),
                 openDocs = Immutable.fromJS(appStore.getOpenDocumentIDs()).filterNot(function (doc) {
```

Comment:
The open document IDs from the application store are now an immutable list, so this immutable conversion is unnecessary.